webapp/channels/src/components/user_settings/general/user_settings_general.tsx

```diff
@@ -319,1088 +119,155 @@ export class UserSettingsGeneralTab extends PureComponent<Props, State> {
-        const {formatMessage} = this.props.intl;
-
-        const active = this.props.activeSection === 'nickname';
-        let max = null;
-        if (active) {
-            const inputs = [];
-
-            let extraInfo;
-            let submit = null;
-            if ((this.props.user.auth_service === 'ldap' && this.props.ldapNicknameAttributeSet) || (this.props.user.auth_service === Constants.SAML_SERVICE && this.props.samlNicknameAttributeSet)) {
-                extraInfo = (
-                    <span>
-                        <FormattedMessage
-                            id='user.settings.general.field_handled_externally'
-                            defaultMessage='This field is handled through your login provider. If you want to change it, you need to do so through your login provider.'
-                        />
-                    </span>
-                );
-            } else {
-                let nicknameLabel: JSX.Element|string = (
+    }, [dispatch, formatMessage, requireEmailVerification, updateSection]);
+
+    const nameSection = (
+        <NameSection
+            activeSection={activeSection}
+            clientError={clientError}
+            firstName={firstName}
+            lastName={lastName}
+            sectionIsSaving={sectionIsSaving}
+            serverError={serverError}
+            setFirstName={setFirstName}
+            setLastName={setLastName}
+            submitUser={submitUser}
+            updateSection={updateSection}
+            updateTab={updateTab}
+            user={user}
+        />
+    );
+
+    const nicknameSection = (
+        <NicknameSection
+            activeSection={activeSection}
+            clientError={clientError}
+            nickname={nickname}
+            sectionIsSaving={sectionIsSaving}
+            serverError={serverError}
+            setNickname={setNickname}
+            submitUser={submitUser}
+            updateSection={updateSection}
+            user={user}
+        />
+    );
+
+    const usernameSection = (
+        <UsernameSection
+            activeSection={activeSection}
+            clientError={clientError}
+            sectionIsSaving={sectionIsSaving}
+            serverError={serverError}
+            setClientError={setClientError}
+            setServerError={setServerError}
+            setUsername={setUsername}
+            submitUser={submitUser}
+            updateSection={updateSection}
+            user={user}
+            username={username}
+        />
+    );
+
+    const positionSection = (
+        <PositionSection
+            activeSection={activeSection}
+            clientError={clientError}
+            position={position}
+            sectionIsSaving={sectionIsSaving}
+            serverError={serverError}
+            setPosition={setPosition}
+            submitUser={submitUser}
+            updateSection={updateSection}
+            user={user}
+        />
+    );
+
+    const emailSection = (
```

Comment:
is there any reason apart from cosmetic just to have these declared as constants here instead of having in the render. I know we had them previously as well

Reply:
Purely cosmetic. Before it made more sense, since a lot more was being done. I still see some value now, because the final render (specially between lines 255 and 267) stays way clearer. But no strong opinions here. I am happy to try to put it all together and see how it looks.

Reply:
0/5 we can leave it as is